src/openmrs-fetch.ts

```diff
@@ -85,33 +87,45 @@ export function openmrsFetch<T = any>(
        * to help developers understand the problem and debug
        */
 
-      // Attempt to download a response body, if it has one
-      return response.text().then(
-        responseText => {
-          let responseBody = responseText;
-          try {
-            responseBody = JSON.parse(responseText);
-          } catch (err) {
-            // Server didn't respond with json, so just go with the response text string
+      /*
+       *Redirect to given url when redirect on auth failure is enabled
+       */
+      const { redirectAuthFailure } = await getConfig("@openmrs/esm-api");
+      if (
+        redirectAuthFailure.enabled &&
+        redirectAuthFailure.errors.indexOf(response.status) >= 0
+      ) {
+        navigateToUrl(redirectAuthFailure.url);
```

Comment:
Thinking about this more - I don't think navigateToUrl will actually work here. The reason is that navigateToUrl calls `history.pushState()` when you're on the same domain, and `location.href = ` when on a different domain.

In our case, we want `history.pushState` when it's within the SPA and `location.assign()` when it's within the refapp/jsp.

I think something like the following would be the best way forward. I'm open to other solutions that you can think of, too

```js
const redirectUrl = new URL(redirectAuthFailure.url)
if (redirectUrl.pathname.startsWith(window.getOpenmrsSpaBase())) {
  navigateToUrl(redirectAuthFailure.url)
} else {
  location.assign(redirectAuthFailure.url)
}
```

^ This logic might be valuable to extract into a reusable function throughout the application, as people jump back and forth between the JSPs and the SPA. For now, I think we could just put it into esm-api though, and extract it later. What are your thoughts?

Reply:
Oops. I missed to notice it is coming from `single-spa`, though I knew it is. Yup, this is useful util through out going forward.

Reply:
I have updated to check whether it is spa or JSP page & redirect. But I am not using URL as it needs full path with host details.